include/rocksdb/merge_operator.h

```diff
@@ -74,18 +74,28 @@ class MergeOperator {
     return false;
   }
 
+  enum MergeValueType : unsigned char {
+    kDeletion = 0x0,
+    kValue = 0x1,
+    kBlobIndex = 0x2,
+  };
+
   struct MergeOperationInput {
-    explicit MergeOperationInput(const Slice& _key,
+    explicit MergeOperationInput(const Slice& _key, MergeValueType _value_type,
```

Comment:
Despite this changes the API, I think it is fine since only rocksdb itself will construct `MergeOperationInput`.